Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -243,6 +262,17 @@ public object Deserialize(IIonReader reader, Type type)
 
         public object Deserialize(IIonReader reader, Type type, IonType ionType)
         {
+            if (options.AnnotatedIonSerializers != null)
+            {
+                foreach (var valuePair in options.AnnotatedIonSerializers)
+                {
+                    if (reader.HasAnnotation(valuePair.Key))
+                    {

```

Comment:
Is there a method to get the current annotation on reader and looking up that in the options.AnnotatedIonSerializers, rather than looping on it?

Reply:
You have to loop on either the list of annotations or the list of annotated ion serializers.

In this case the list of annotated ion serializers is always the relevant information and the annotations aren't, so I decided to loop on that.